pkg/handler/dencode.go

```diff
@@ -1,126 +1,77 @@
 package handler
```

Comment:
Could you please write here some unit tests for functions dencode and readSecret?

Reply:
sure sure